stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -1631,9 +1631,13 @@ extension ${Self} {
   public init?(exactly value: ${That}) {
     _value = Builtin.${sign}itofp_${ThatBuiltinName}_FPIEEE${bits}(value._value)
 
-%   if srcBits >= SignificandBitCount:
-    guard let roundTrip = ${That}(exactly: self),
-      roundTrip == value else {
+%   if srcBits > SignificandBitCount:
```

Comment:
This check isn't quite as fine-grained as it should be, though it doesn't matter for `Float` or `Double` and any of the usual integer sizes. Binary floating-point can represent any integer with `<= SignificandBitCount + 1` bits (+1 because of the implicit bit), and `srcBits` doesn't account for signed/unsigned.

Reply:
Thanks. Should I also make the code resilient to short exponents that cannot encode the full range of the integer type?

Reply:
If it's useful for reference, I implemented some of the routines for generic conversion in the other direction (integer to floating-point) as well as floating-point to floating-point:  

https://github.com/apple/swift/pull/13782